server/backbone/api/parent.py

```diff
@@ -542,3 +558,70 @@ def get_child_quests(cid, start, lookahead):
     if not child_is_my_child(child):
         raise BackboneException(404, "Child not found")
     return json_return(get_childs_quest_with_window(start, lookahead))
+
+
+@api_bp.route('/quest/<int:qid>/verify', methods=['POST'], defaults={'ts': None})
+@api_bp.route('/quest/<int:qid>/<float:ts>/verify', methods=['POST'])
+@parent_login_required
+@backbone_error_handle
+def verify_quest_completion(qid, ts):
+    """
+    .. :quickref: quest; verify a quest completion
+
+    Verify a quest that has been completed.
+
+    **Parent login required**
+
+    **errors**:
+
+    404, quest not found - quest does not exists, or not currently logged in child's
+
+    **example return**:
+
+    .. code-block:: json
+
+        {
+          "data": {
+            "lvled_up": true,
+            "qst": {
+              "completed_on": "Sat, 06 Jul 2019 21:30:53 GMT",
+              "description": "You're going on a quest to save the princess, brush your teeth so you don't embarrass yourself.",
+              "due": 1559250000.0,
+              "id": 2,
+              "needs_verification": true,
+              "recurring": false,
+              "reward": 12,
+              "title": "This is the initial quest",
+              "verified_on": "Sat, 06 Jul 2019 21:34:36 GMT"
+            },
+            "verified_now": true
+          }
+        }
+
+    :param qid: id of quest to be completed
+    :param ts: timestamp the quest needs to be completed by
+    :return: quest description with ``lvled_up`` and ``completed_now`` filds added
+    """
+    quest = Quest.query.filter_by(id=qid).first()
+    quest_owner = Child.query.filter_by(id=quest.owner).first()
+    if not child_is_my_child(quest_owner):
+        raise BackboneException(404, "Quest not found")
+    if not ts:
+        ts = find_next_time(quest)
+    else:
+        ts = datetime.fromtimestamp(ts)
+    previous_verification = QuestVerifications.query.filter_by(id=qid, value=ts).first()
+    verified_now = previous_verification is None
+    lvl_up = False
+    if verified_now:
+        # Not already verified
```

Comment:
I'm confused does `verified_now == True` mean the quest is not verified??

Reply:
It means that this call verified the quest and not a previous call